### DISCUSSION TITLE
Feature request: return the number of keys in a collection with a filter

### DISCUSSION THREAD

**Discussion by theMackabu** [pocketbase/pocketbase#249] [Q&A] [answered]
Is it possible to get a sum of users profile with a count?
![Screen Shot 2022-08-01 at 9 59 33 AM](https://user-images.githubusercontent.com/32078755/182078490-315964cf-fe5f-489b-a197-f083926b8dc3.png)
so for example, if I post /api/profile/sales, and 3 users have 1 sale each, it would return 3
would this be viable or much faster to just add them together in the frontend?

**Comment by theMackabu** [comment, marked as answer]
Turns out its faster to compute on the frontend, nevermind about this question

**Reply by theMackabu** [reply]
It would still be cool if there was a way to only return number of keys with the filter api

**Reply by ganigeorgiev** [reply]
I'm not sure that I understand your use case. Could you elaborate a little more?
Aggregations and reporting is usually left to the users to do on their own using the internal DB abstraction.

**Reply by theMackabu** [reply]
Something like the opposite of expand param, where a relation only shows the count of objects in it just to save on request speed? (if it has a lot of relations)

**Reply by ganigeorgiev** [reply]
But you already have that "count".

If you have a relation multi-value field you can always call `.length` on the value array - `yourUser.profile.sales.length`.

If `sales` in your case is a collection and you want to return the number of all sales for all users that matches a specific filter, you can just call the list action with `perPage=1`, eg.:
```
const result = await client.Records.getList("sales", 1, 1);
console.log(result.totalItems);
```

**Reply by theMackabu** [reply]
Ah, so I can have pages on relations too, quite cool
